src/detect-sid.c

```diff
@@ -67,7 +67,7 @@ static int DetectSidSetup (DetectEngineCtx *de_ctx, Signature *s, const char *si
         SCLogError("sid value 0 is invalid");
         goto error;
     }
-    if (s->id > 0) {
+    if (s->id > 0 && s->id != id) {
```

Comment:
will this also allow `sid:1; sid:1;` in a rule? If so, I'm not really in favor of it.